ide/src/tests/install/install.test.ts

```diff
@@ -0,0 +1,22 @@
+import { suite, test } from "mocha";
+import { expect } from "chai";
+import vscode from "vscode";
+import waitForExpect from "wait-for-expect";
+
+const flowistryCommandsExist = async () => {
+    const commands = await vscode.commands.getCommands();
+    return commands.filter((command) => command.includes('flowistry.')).length > 0;
+};
+
+suite("Flowistry installation tests", () => {
```

Comment:
Is there something you'd like to test in addition to the Flowistry commands being available? I was thinking either:

- Expect `cargo flowistry -V` returns a valid version and doesn't fail
- Run a Flowistry command in VSCode and expect it doesn't fail

Reply:
Would it also be valuable to have a test that blocks requests to the release binaries causing the extension to compile from source?

Reply:
My first instinct was that both of those things should already be tested by running the slicer. But I see that those tests don't get executed in the `install_head` / `install_release` jobs.

Is the reason we don't just do `npm test` in those jobs for efficiency?

Reply:
> Would it also be valuable to have a test that blocks requests to the release binaries causing the extension to compile from source?

It's enough of an edge case that we don't need to worry about it for now.

Reply:
> Is the reason we don't just do npm test in those jobs for efficiency?

I initially separating them to:

- Make it quicker to run tests by
    - skipping testing the latest release since it should already have been tested and
    - running the `install` and `slice` tests in parallel
- Make it clear whether the extension failed because it installed incorrectly or because the extension/driver is broken

That being said, it might be beneficial to test after install since I could remove `test-vscode` and run the tests after install. Another benefit would be removing the `install-build` composite action thingy since `install_head` and `install_release` will do that for us.

Whatever seems best works for me!

Reply:
For now, I think let's just keep it simple. Let's do one job in the following sequence:

1. `cargo build`
2. copy binaries into a distribution zip
3. `cargo test`
4. `npm build`
5. `npm test --install <the zip>`
 
This loses the parallelism between cargo and vscode tests, but that shouldn't be a big issue because the cargo tests (at least) are very fast. Certainly faster than spinning up a new VM. I don't think testing vscode with pre-installed binaries is important -- just testing the "install from server" path is fine.